tests/test_proxies.py

```diff
@@ -23,25 +23,65 @@ def request_get(port, path, token, host='localhost'):
 def test_server_proxy_minimal_proxy_path_encoding():
     """Test that we don't encode anything more than we must to have a valid web
     request."""
-    special_path = quote("Hello world 123 åäö 🎉你好世界±¥ :/[]@!$&'()*+,;=-._~", safe=":/?#[]@!$&'()*+,;=-._~")
-    # NOTE: we left out ?# as they would interact badly with our requests_get
-    # function's ability to pass the token query parameter.
+    special_path = quote("Hello world 123 åäö 🎉你好世界±¥ :/[]@!$&'()*+,;=-._~?key1=value1", safe=":/?#[]@!$&'()*+,;=-._~")
     test_url = '/python-http/' + special_path
     r = request_get(PORT, test_url, TOKEN)
     assert r.code == 200
     s = r.read().decode('ascii')
-    assert 'GET /{}?token='.format(special_path) in s
+    assert 'GET /{}&token='.format(special_path) in s
```

Comment:
This test was already a bit confusing due to the long list of unicode characters. Could you explain why the assertion needs to change? The related issue mentions `test_server_proxy_minimal_proxy_path_encoding_complement`.

Reply:
the request_get() helper function returns the response to a request that is appended with the token as a query string, but the token is appended with `?token=...` or `&token=...` depending on if it found a `?` within the URL it is making a request to as it needs to do in order to not define the query params to start a second time, as indicated by the first `?` found in the URL.

Reply:
What's the reason for changing the URL in this test, as opposed to adding a new test? Are we losing a test case by changing it?

Reply:
No, just making it more explicit that ? won't be escaped on arrival to the receiving end.